packages/metro/src/node-haste/DependencyGraph.js

```diff
@@ -73,7 +73,9 @@ class DependencyGraph extends EventEmitter {
     this._createModuleResolver();
   }
 
-  static _createHaste(config: ConfigT): JestHasteMap {
+  static _createHaste(config: ConfigT, watch?: boolean): JestHasteMap {
+    const shouldWatch = watch === undefined ? !ci.isCI : watch;
```

Comment:
Could you inline this below instead of using the variable? It's a simple expression. Also, let's use `watch == null` instead of checking for undefined.